[4782] Add <- and -> (forward and back) buttons to the URL/internal browser in Preview Toolbar

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4782

aachavarria committed:
Moving PreviewForwardButton next to PreviewBackButton
Updating icon;

aachavarria committed:
adding historyNavigationType to determinate how to modify history stacks

rart committed:
Update GlobalState.ts

sumerjabri commented:
@rart I'll wait for the final updates we did today to make it in (I wasn't able to find it in this PR).

aachavarria committed:
Creating New Reducer for PreviewNavigation;
Moving actions/reducer to previewNavigation.ts
renaming currentUrl to currentFullUrl;
renaming computedUrl to currentUrlPath;

rart commented on ui/app/src/state/reducers/previewNavigation.ts:
```diff
@@ -0,0 +1,129 @@
+    historyNavigationType: null
+  },
+  {
+    [checkInGuest.type]: (state, { payload }) => {
```
Please add these type of comments to each action
```suggestion
    // A page is being visited or reloaded
    // - Push the url into the back stack (if not already pushed)
    // - Clear the forward stack (if not already pushed)
    // - Update the current url
    [checkInGuest.type]: (state, { payload }) => {
```

rart replied:
Still missing the comments for the rest of the actions

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4782

 Conflicts:
	ui/app/src/state/reducers/preview.ts

aachavarria committed:
Mergin dev

aachavarria committed:
Removing currentFullUrl

rart commented on ui/guest/src/react/GuestProxy.tsx:
```diff
@@ -307,9 +307,9 @@ export function GuestProxy() {
+              const { modelId, fieldId, targetIndex, currentFullUrl } = payload;
               let ifrm = document.createElement('iframe');
-              ifrm.setAttribute('src', `${currentUrl}`);
+              ifrm.setAttribute('src', `${currentFullUrl}`);
```
```suggestion
              ifrm.setAttribute('src', currentFullUrl);
```

rart commented on ui/app/src/state/reducers/previewNavigation.ts:
```diff
@@ -0,0 +1,112 @@
+      }
+
+      return nextState;
+    }
```
```suggestion
    [changeSite.type]: (state, { payload }) => ({
      ...state,
      historyBackStack: [],
      historyForwardStack: [],
      historyNavigationType: null,
      currentUrlPath: payload.nextUrl ?? state.currentUrlPath
    })
```

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -629,7 +631,7 @@ export function PreviewConcierge(props: any) {
               hostToGuest$.next({
                 type: INSERT_OPERATION_COMPLETE,
-                payload: { ...payload, currentUrl }
+                payload: { ...payload, currentFullUrl: guestBase + currentUrlPath }
```
```suggestion
                payload: { ...payload, currentFullUrl: ˋ${guestBase}${currentUrlPath}ˋ }
```

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -588,7 +590,7 @@ export function PreviewConcierge(props: any) {
               hostToGuest$.next({
                 type: INSERT_OPERATION_COMPLETE,
-                payload: { ...payload, currentUrl }
+                payload: { ...payload, currentFullUrl: guestBase + currentUrlPath }
```
```suggestion
                payload: { ...payload, currentFullUrl: ˋ${guestBase}${currentUrlPath}ˋ }
```

rart commented on ui/app/src/modules/Preview/Host.tsx:
```diff
@@ -192,7 +195,7 @@ export default function Host() {
     >
       <HostUI
-        url={currentUrl}
+        url={currentUrlPath === '' ? previewLandingBase : guestBase + currentUrlPath}
```
```suggestion
        url={currentUrlPath === '' ? previewLandingBase : ˋ${guestBase}${currentUrlPath}ˋ}
```

aachavarria committed:
adding comments to checkInGuest

aachavarria committed:
removing not used imports

aachavarria committed:
adding comments to previewNavigation.ts reducer
